src/Microsoft.AspNet.OData.Shared/PatchMethodHandler.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.OData
+{    
+    /// <summary>
+    /// Basic Interface for PatchMethodHAndler
+    /// </summary>
+    public interface IPatchMethodHandler
```

Comment:
>IPatchMethodHandler [](start = 21, length = 19)

This should define TryCreate, TryGet, TryDelete, GetNestedPatchHandler -- otherwise, what is the use of having the interface?

Reply:
The abstract class is a generic class, and the methods use Tstructural type. But in GetNestedPatchHandler we need to return the patch handler dynamically, so for those uses uses interface